### PR TITLE
Implements item actions and deletes weapons

### LINKED ISSUE #29: Item actions
In addition to modifiers, items have actions

Examples:

* A longsword grants the attack action of a 1d8 slashing damage
* A greatsword grants the attack action of 2d6 slashing damage
* A potion of healing grants the heal action of 2d4 + 2
* Ring of the ram grants the attack action of 2d10 force damage with +7 to the attack roll. This also pushes the enemy 5 feet away and the ring only has 3 charges, regaining 1-3 charges each day.

Note that the attack roll bonus for weapons is your proficiency bonus plus your strength modifier (or your dexterity modifier for finesse weapons if your dex is greater than your strength)

So I believe that items should have a 1 to many link to actions (some items grant several options)

| item_id | action_type | dice_id | dice_count | action_bonus | roll_bonus | description | consumes_item |  damage_type_id  |
|-------|-------------|--|--|--|--|--|--|--|
| 1 | attack | 3 | 1 | 0 | 0 | null | false | 2 |
| 2 | attack | 2 | 2 | 0 | 0 | null | false | 2 |
| 3 | heal | 1 | 2 | 2 | 0 | null | true | null |
| 4 | attack | 4 | 2 | 0 | 7 | On success, move target back 5 feet. | false | 6 |

I'm not sure how best to handle the charges that some items have

Actions can also be attached to spells (which I think are distinct from items) and races/classes among others  (probably)

### PR DESCRIPTION
**Pull request by G-Bro**
Resolves #29

### CODE REVIEW

File: app/Http/Resources/MonsterResource.php [hunk 1]
```diff
@@ -104,21 +104,27 @@ public function toArray($request)
...
+                                        'damage' => [
+                                            'dice' => $action->dice->sides,
+                                            'dice_count' => $action->dice_count,
+                                            'as_string' => $action->dice_as_string()
+                                        ],
+                                    ];
+                                }
+                            ),
+                        ];
+                    }
```
Comment (DanEvans485) [inline app/Http/Resources/MonsterResource.php]: Can these be a WeaponResource and ActionResource ?

**Comment by G-Bro** [comment]
[opendm - hi.txt](https://github.com/G-Bro/OpenDM-Server/files/4177214/opendm.-.hi.txt)

File: app/Http/Resources/MonsterResource.php (hunk 1 above)
Comment (G-Bro) [reply app/Http/Resources/MonsterResource.php]: yes

Comment (G-Bro) [reply app/Http/Resources/MonsterResource.php]: @DanEvans485 I've done that and a few others too

**Pull request merged**